cri/v1alpha2/cri.go

```diff
@@ -1161,7 +1189,30 @@ func (c *CriManager) UpdateContainerResources(ctx context.Context, r *runtime.Up
 // to either create a new log file and return nil, or return an error.
 // Once it returns error, new container log file MUST NOT be created.
 func (c *CriManager) ReopenContainerLog(ctx context.Context, r *runtime.ReopenContainerLogRequest) (*runtime.ReopenContainerLogResponse, error) {
-	return nil, fmt.Errorf("ReopenContainerLog Not Implemented Yet")
+	containerID := r.GetContainerId()
+
+	container, err := c.ContainerMgr.Get(ctx, containerID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get container %q with error: %v", containerID, err)
+	}
+	if !container.IsRunning() {
+		return nil, fmt.Errorf("container is not running")
```

Comment:
I was thinking that CRI should use the right status code for the error. For this one, we should use ErrPreconditionFail something like that. How do you think about this?

Reply:
Yeah, Agree with that. THX.